### ISSUE TITLE
Feature Request: Updates

### ISSUE THREAD

**Issue by nutterthanos** [httptoolkit/httptoolkit#146]
Maybe updates could be done with https://github.com/Squirrel/Squirrel.Windows just like with Atom/Discord etc
---

_Does this affect you too? Click the +:smiley: icon in the top right and add a :+1: on issues to help decide where HTTP Toolkit goes next, or [go take a look & vote on the most popular ideas so far](https://github.com/httptoolkit/feedback/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc)._

**Comment by pimterry** [comment]
Hi @nutterthanos - this is something that could be improved further, but HTTP Toolkit does automatically update all its functionality already today. The only bit that doesn't autoupdate is [the desktop shell](https://github.com/httptoolkit/httptoolkit-desktop/), which doesn't do very much.

More specifically, the [web UI](https://github.com/httptoolkit/httptoolkit-ui/) updates by using a service worker to manage cached versions of the app, and [the backend](https://github.com/httptoolkit/httptoolkit-server/) updates using https://github.com/oclif/plugin-update.

The shell doesn't update, but it also doesn't include very much at all: just the logic to start the server, and open the UI from the web, which doesn't really change very often!

Splitting out updates like this makes them dramatically smaller. Rather than pulling down a whole new Electron build, you only have to download the small specific component that's changed. The most common changes are in the UI, and updating that is generally <1MB (due to bundle splitting) or might be up to 10MB if some large dependencies change. The latest server is about 30MB. Meanwhile the full Electron app is 80 - 120MB.

This also means we use exactly the same update mechanism for all platforms, and for users who want to use HTTP Toolkit directly in a browser running the server manually to avoid any Electron overhead. Meanwhile Squirrel works very differently for Windows & Mac, and as far as I can tell there are no auto-update systems for deb/rpm packages at all.

There's definitely room for improvement here, since there are occasional changes to the desktop shell that it'd be nice to easily distribute, but they're not a huge problem. It's possible to check the desktop shell version from the UI too, so if there were ever a super important update, I'd push a check to the UI which prompted everybody to manually update the desktop app.

That said, if you are interested in this I'd happily accept a PR to [the desktop shell](https://github.com/httptoolkit/httptoolkit-desktop/) to enable that. That's built using [Electron Builder](https://www.electron.build/auto-update) which has some support for this built into the NSIS & DMG targets, which we already use for Windows & Mac. Remember that [HTTP Toolkit Pro is totally free for contributors :smiley:](https://github.com/httptoolkit/httptoolkit#contributing-directly)

**Comment by nutterthanos** [comment]
yeah with linux you would just push the new builds to the repos that exist in ubuntu/centos etc or have your own depending on needs
